.github/workflows/impersonator.yml

```diff
@@ -0,0 +1,72 @@
+name: Impersonator
+
+env:
+  PRERELEASE_BRANCHES: alpha,beta,rc
+  DOCKER_HUB_REPO: dolittle/impersonator
+
+on:
+  push:
+    branches:
+    - '**'
+  pull_request:
+    types: [closed]
+
+jobs:
+  build:
+    name: Build + Test
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-go@v2
+    - name: Build
+      working-directory: ./Source/Impersonator
+      run: go build .
+    - name: Test
+      working-directory: ./Source/Impersonator
+      run: go test -v ./...
+  
+  publish:
+    name: Publish
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Establish context
+      id: context
+      uses: dolittle/establish-context-action@v2
+      with:
+        prerelease-branches: ${{ env.PRERELEASE_BRANCHES }}
+    - uses: docker/setup-buildx-action@v1
+      if: ${{ steps.context.outputs.should-publish == 'true' }}
+    - uses: docker/login-action@v1
+      if: ${{ steps.context.outputs.should-publish == 'true' }}
+      with:
+        username: ${{ secrets.DOCKER_USERNAME }}
+        password: ${{ secrets.DOCKER_PASSWORD }}
+    - name: Build Image
+      uses: docker/build-push-action@v2
+      if: ${{ steps.context.outputs.should-publish == 'true' && env.DOCKER_HUB_REPO != '' }}
+      with:
+        push: false
+        context: ./Source/Impersonator
+        file: ./Source/Impersonator/Dockerfile
+    - name: Increment version
+      id: increment-version
+      if: ${{ steps.context.outputs.should-publish == 'true' }}
+      uses: dolittle/increment-version-action@v2
+      with:
+        version: ${{ steps.context.outputs.current-version }}
+        release-type: ${{ steps.context.outputs.release-type }}
+    - name: Create GitHub Release
+      uses: dolittle/github-release-action@v1
+      if: ${{ steps.context.outputs.should-publish == 'true' }}
+      with:
+        cascading-release: ${{ steps.context.outputs.cascading-release }}
+        version: ${{ steps.increment-version.outputs.next-version }}
```

Comment:
Should it create a GitHub release?

And the cascading-release part should not be there

Reply:
Good question - I will feed that directly back to you. I was wondering if it would be problematic if this GitHub release thing is triggered from multiple jobs (as it is now)? Maybe we should move the Docker-building and creating release for all three images into one job?

Reply:
Well the question essentially boils down to how the repository is versioned. As this is now a kind of "mono repo" h how should the different applications be versioned?

Reply:
To your question: Yes, it would be a trouble if triggered from different jobs 😋

Reply:
So that should be fixed now.